src/Storages/StorageMongoDB.cpp

```diff
@@ -43,6 +45,8 @@ StorageMongoDB::StorageMongoDB(
     , collection_name(collection_name_)
     , username(username_)
     , password(password_)
+    , options(options_)
+    , uri("mongodb://" + username_ + ":" + password_ + "@" + host_ + ":" + std::to_string(port_) + "/" + database_name_ + "?" + options_)
```

Comment:
I'm a bit confused - how username/password was passed before your changes?

Reply:
In line 68 there is an if-condition to authenticate the connection in case it is not.
I just checked and we can omit adding the username and password to the URI.